test/space/ntuples_test.py

```diff
@@ -346,112 +347,76 @@ def test_power(fn):
     assert all_almost_equal(y_pos, y_pos_arr)
 
 
-def _test_unary_operator(fn, function):
-    # Verify that the statement y=function(x) gives equivalent results
-    # to NumPy
-    x_arr, x = example_vectors(fn)
+def test_unary_ops(fn):
+    # Verify that the unary operators (`+x` and `-x`) work as expected
 
-    y_arr = function(x_arr)
+    for op in [operator.pos, operator.neg]:
+        x_arr, x = example_vectors(fn)
 
-    y = function(x)
+        y_arr = op(x_arr)
+        y = op(x)
 
-    assert all_almost_equal([x, y], [x_arr, y_arr])
+        assert all_almost_equal([x, y], [x_arr, y_arr])
 
 
-def _test_binary_operator(fn, function):
-    # Verify that the statement z=function(x,y) gives equivalent results
+def _test_scalar_operator(fn, function):
+    # Verify that the statement y=op(x, scalar) gives equivalent results
     # to NumPy
+
+    for scalar in [-31.2, -1, 0, 1, 2.13]:
+        x_arr, x = example_vectors(fn)
+
+        if scalar == 0 and function in [operator.truediv,
+                                        operator.itruediv]:
+            # Check for correct zero division behaviour
+            with pytest.raises(ZeroDivisionError):
+                y = function(x, scalar)
+        else:
+            y_arr = function(x_arr, scalar)
+            y = function(x, scalar)
+
+            assert all_almost_equal([x, y], [x_arr, y_arr])
+
+        # right op
+        x_arr, x = example_vectors(fn)
+
+        y_arr = function(scalar, x_arr)
+        y = function(scalar, x)
+
+        assert all_almost_equal([x, y], [x_arr, y_arr])
```

Comment:
the other layer of indirection is removed in the latest commit. Other than that I'll write some comment.